aptos-move/framework/aptos-framework/sources/configs/staking_config.move

```diff
@@ -146,7 +149,10 @@ module aptos_framework::staking_config {
             new_rewards_rate_denominator > 0,
             error::invalid_argument(EZERO_REWARDS_RATE_DENOMINATOR),
         );
-
+        assert!(
+            new_rewards_rate < 100,
```

Comment:
This is actually not quite right. It should be new_rewards_rate < new_rewards_rate_denominator. The rewards rate is a fraction represented by new_rewards_rate as the numerator and new_rewards_rate_denominator

Reply:
100 comes from the previous comment.
> I wonder if we just make assumptions that stake_amount < u64.max, rewards_rate < 100 and num_successful_proposals < 86400 (1 proposal per second in a day) and just have an ensure that the reward calc doesn't overflow?

Reply:
However, `new_rewards_rate < new_rewards_rate_denominator` sounds necessary as well. How about `new_rewards_rate < 100 && new_rewards_rate < new_rewards_rate_denominator`?

Reply:
By the way, should it be `new_rewards_rate < new_rewards_rate_denominator` or `new_rewards_rate <= new_rewards_rate_denominator` (i.e., `<` vs. `<=`).

Reply:
Added the following comment, and resolved it:
```
// `rewards_rate` which is the numerator is limited to be less than 100 in order to avoid the arithmetic
// overflow in the rewards calculation. `rewards_rate_denominator` can be adjusted to get the desired rewards
// rate (rewards_rate / rewards_rate_denominator).
```

Reply:
Actually, I take this assertion out of this PR. Limiting `rewards_rate` to be less than 100 breaks a lot of the current e2e tests. It should be another separate PR if we really want to limit its upper bound. (@movekevin )